lib/google_crawler/search/keyword_worker.ex

```diff
@@ -0,0 +1,28 @@
+defmodule GoogleCrawler.Search.KeywordWorker do
+  use Oban.Worker,
+    priority: 1,
+    max_attempts: 10,
+    unique: [period: 30]
+
+  alias GoogleCrawler.Repo
+  alias GoogleCrawler.Search.{Keyword, Parser}
+
+  @impl Worker
+  def perform(%Oban.Job{args: %{"keyword_id" => keyword_id}}) do
+    keyword = Repo.get_by!(Keyword, %{id: keyword_id})
+
+    updated_keyword =
+      keyword
+      |> Ecto.Changeset.change(status: :in_progress)
+      |> Repo.update!()
+      |> Ecto.Changeset.change(result: parsed_result(keyword), status: :finished)
```

Comment:
@rosle Cannot use cast_assoc, as it is not a association but a embed 

<img width="1948" alt="Screen Shot 2564-06-08 at 11 33 26" src="https://user-images.githubusercontent.com/11224179/121123425-6788fd00-c84d-11eb-8be8-b9d7bf8846ee.png">

Reply:
Oh I think we can check `cast_embed` instead 👍

Reply:
Yea, I tried it before as well, but we need to use `put_embed` as we are updating. Updated in https://github.com/ankitkalia1195/elixir-google-crawler/pull/41/commits/52e35a147f922c2751c1fa0e0cae52211b7de331

Reply:
Oh ok, I thought it will require changeset on `Search.Result` to validate. Probably the one I tried is from map to embedded schema but here we've already build a `Search.Result` in the parser 🤔 💭